feat: add isThemeEditorDrawerOpen prop

panbenson opened the pull request:
add the `isThemeEditorDrawerOpen` prop to `ThemeEditor` for more customizability

verified changes in storybook.

panbenson commented:
@it-nalon could you take a look at these changes when you have a chance?

it-nalon commented:
Hi, sorry for the delay!
I checked the code and it seems a good idea to control the open/close state outside the component.

At this point, I think we can simply make the `<ThemeEditor />` component completed controlled.
It would be awesome if you can edit the PR so it accepts all the: `UseDisclosureProps` props.

All these props could then be passed to the: `useDisclosure()` hook from Chakra UI.

The could could look something like this:
```tsx

...

export type ThemeEditorProps = {
  children: ThemeEditorChild[] | ThemeEditorChild
} & UseDisclosureProps

export const ThemeEditor: FC<ThemeEditorProps> = ({ children, ...disclosureProps }) => {
  const { isOpen, onOpen, onClose } = useDisclosure(disclosureProps)
  
 ...

}
```

With this changes you should be able to control the `ThemeEditor` state passing just the `isOpen` props:
```tsx
const [isOpen, setIsOpen] = useState(true)

<ThemeEditor isOpen={isOpen} />
```

it-nalon commented on packages/chakra-ui-core/src/components/theme-editor/ThemeEditor.tsx:
```diff
@@ -11,10 +11,17 @@ type ThemeEditorChild = React.ReactElement<
-export const ThemeEditor: FC<ThemeEditorProps> = ({ children }) => {
-  const { isOpen, onOpen, onClose } = useDisclosure()
+export const ThemeEditor: FC<ThemeEditorProps> = ({ children, isThemeEditorDrawerOpen }) => {
+  const [isOpen, setIsOpen] = useState(isThemeEditorDrawerOpen)
```
We should not re-implement functionalities that are already available from Chakra UI.
I think we can just use the `useDisclosure()` hook and pass to it the `isOpen`, `onClose`, etc. props.

it-nalon commented on packages/chakra-ui-core/src/components/theme-editor/ThemeEditor.tsx:
```diff
@@ -11,10 +11,17 @@ type ThemeEditorChild = React.ReactElement<
+export const ThemeEditor: FC<ThemeEditorProps> = ({ children, isThemeEditorDrawerOpen }) => {
+  const [isOpen, setIsOpen] = useState(isThemeEditorDrawerOpen)
+  const onOpen = () => setIsOpen(true)
+  const onClose = () => setIsOpen(false)
```
Please avoid non-memoized functions, we accept code only if it is properly memoized.

Instead of: `const onOpen = () => setIsOpen(true)` you can write: `const onOpen = useCallback(() => setIsOpen(true), [])` and it will be accepted.

Many thanks!

panbenson commented:
thanks for being open to this contribution @it-nalon! I incorporated your suggestions in my most recent commits, please let me know if there's anything else

I also added a story just to demo:
https://user-images.githubusercontent.com/15132025/154892685-ef1931ae-2b51-4703-83e2-67252ce0147e.mov

it-nalon commented:
Hi @panbenson, I've merged the PR and it will be released on the 0.2.0 version (because you pointed the PR to the main branch), the 0.2.0 does not have a release date yet.

I think that this PR could be deployed as version: 0.1.4 before the 0.2.x version.
If you make a PR that points to the `support/0.1.x` branch I will deploy a new 0.1.x version as soon as possible.
